source/SH3/system/glprogram.cpp

```diff
@@ -75,11 +80,17 @@ GLuint sh3_glprogram::Compile(std::uint32_t type)
 
         source.open(fname);
 
+        if(!source.is_open())
+            die("sh3_glprogram::Compile( ): Unable to open a handle to %s!", fname.c_str());
+
         while(!source.eof())
```

Comment:
Safer as in, doesn't lead to errors when using different line endings.
You should be able to switch to different line endings in your editor.